cylc/flow/task_pool.py

```diff
@@ -128,7 +128,7 @@ def __init__(
         self.task_name_list = self.config.get_task_name_list()
         self.task_queue_mgr = IndepQueueManager(
             self.config.cfg['scheduling']['queues'],
-            self.config.get_task_name_list(),
+            self.task_name_list,
```

Comment:
what's the difference between this and `self.config.taskdefs` - is this the cached version?

How do you know when to use each?